addons/account/tests/test_invoice_taxes.py

```diff
@@ -427,3 +427,48 @@ def test_misc_journal_entry_tax_tags_purchase(self):
             {'balance': -100.0,     'tax_ids': [],              'tag_ids': self.tax_tag_pos.ids},
             {'balance': 1100.0,     'tax_ids': [],              'tag_ids': []},
         ])
+
+    def test_tax_calculation_foreign_currency_large_quantity(self):
+        ''' Test:
+        Foreign currency with rate of 1.1726 and tax of 21%
+        price_unit | Quantity  | Taxes
+        ------------------
+        2.82       | 20000     | 21% not incl
+        '''
+        self.env['res.currency.rate'].search([]).unlink()
+        self.currency_usd_id = self.env.ref("base.USD")
+        self.currency_chf_id = self.env.ref("base.CHF")
+        # make sure that we have USD as base...
+        self.cr.execute("UPDATE res_company SET currency_id = %s WHERE id = %s", 
+            [self.currency_usd_id.id, self.env.company.id])
+        self.env['res.currency.rate'].create({
```

Comment:
This test is terribly ugly... You need to use the generic accounting setup instead of hacking the current company like this.

Reply:
We only learn from the best ;)
https://github.com/odoo/odoo/commit/6deed451ee686d104583cb33d66c5fbc400cad8d

Reply:
.... "5 Mar 2018"

Reply:
<cricket noise>